datafusion/functions/benches/to_timestamp.rs

```diff
@@ -0,0 +1,116 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+extern crate criterion;
+
+use criterion::{black_box, criterion_group, criterion_main, Criterion};
+
+use datafusion_expr::lit;
+use datafusion_functions::expr_fn::to_timestamp;
+
+fn criterion_benchmark(c: &mut Criterion) {
+    c.bench_function("to_timestamp_no_formats", |b| {
+        let inputs = vec![
+            lit("1997-01-31T09:26:56.123Z"),
+            lit("1997-01-31T09:26:56.123-05:00"),
+            lit("1997-01-31 09:26:56.123-05:00"),
+            lit("2023-01-01 04:05:06.789 -08"),
+            lit("1997-01-31T09:26:56.123"),
+            lit("1997-01-31 09:26:56.123"),
+            lit("1997-01-31 09:26:56"),
+            lit("1997-01-31 13:26:56"),
+            lit("1997-01-31 13:26:56+04:00"),
+            lit("1997-01-31"),
+        ];
+        b.iter(|| {
+            for i in inputs.iter() {
+                black_box(to_timestamp(vec![i.clone()]));
```

Comment:
The benchmarks had to change to test the public api vs the previous version that tested the internal api.

Reply:
I think testing the public API is actually a more real-world comparison (so this seems like an improvement to me)